h2o-py/h2o/backend/connection.py

```diff
@@ -35,14 +35,129 @@
 from h2o.utils.typechecks import (assert_is_type, assert_matches, assert_satisfies, is_type, numeric)
 from h2o.model.metrics_base import (H2ORegressionModelMetrics, H2OClusteringModelMetrics, H2OBinomialModelMetrics,
                                     H2OMultinomialModelMetrics, H2OAutoEncoderModelMetrics)
+from h2o.utils.typechecks import assert_is_type, is_type
 
-__all__ = ("H2OConnection", )
+__all__ = ("H2OConnection", "H2OConnectionConf", )
 
 if tuple(int(x) for x in requests.__version__.split('.')) < (2, 10):
     print("[WARNING] H2O requires requests module of version 2.10 or newer. You have version %s.\n"
           "You can upgrade to the newest version of the module running from the command line\n"
           "    $ pip%s install --upgrade requests" % (requests.__version__, sys.version_info[0]))
 
+class H2OConnectionConf(object):
+    """
+    Configuration of connection to H2O.
+
+    The main goal of this class is to specify location of running
+    H2O instance and properties of connection. The location of running instance
+    is given by schema, ip, port, context_path parameters forming connection URL in
+    the following way: ``schema://ip:port/context_path``
+    """
+
+    @staticmethod
+    def create(conf):
+        """
+        Create a new instance of this class from given dictionary.
+        The method only uses keys which corresponds to properties
+        this class, ignores rest of keys.
+        :param conf:  dictionary of parameters
+        :return:  a new instance of this class filled with values from given dictionary
+        """
+        cconf = H2OConnectionConf()
+        for k,v in conf.items():
+            if hasattr(cconf, k):
```

Comment:
Using `hasattr` in this context is not the best choice, because there are many attrs on this class, inherited from Object, and we don't want to make them overridable. It's easier to declare local `allowed_params = {"ip", "port", "https", ...}` and check whether `k` is in this set.